app/src/endpoints.c

```diff
@@ -233,18 +300,21 @@ static void disconnect_current_endpoint() {
     zmk_endpoints_send_report(HID_USAGE_CONSUMER);
 }
 
-static void update_current_endpoint() {
-    enum zmk_endpoint new_endpoint = get_selected_endpoint();
+static void update_current_endpoint(void) {
+    struct zmk_endpoint_instance new_instance = get_selected_instance();
 
-    if (new_endpoint != current_endpoint) {
-        /* Cancel all current keypresses so keys don't stay held on the old endpoint. */
+    if (!zmk_endpoint_instance_equals(new_instance, current_instance)) {
+        // Cancel all current keypresses so keys don't stay held on the old endpoint.
         disconnect_current_endpoint();
 
-        current_endpoint = new_endpoint;
-        LOG_INF("Endpoint changed: %d", current_endpoint);
+        current_instance = new_instance;
+
+        char endpoint_str[10];
```

Comment:
Can we have a `#define` with the nominal length `zmk_endpoint_instance_print` needs (similar to [`BT_ADDR_LE_STR_LEN` for `bt_addr_to_str`](https://github.com/zephyrproject-rtos/zephyr/blob/706cfbbb76151f4223c225c98295a44c8ecec598/include/zephyr/bluetooth/addr.h#L206-L225))?

Reply:
Done.